playground/legacy/__tests__/legacy.spec.ts

```diff
@@ -52,22 +52,22 @@ test('generates assets', async () => {
     () => page.textContent('#assets'),
     isBuild
       ? [
-          'index: 404',
-          'index-legacy: 404',
-          'chunk-async: 404',
-          'chunk-async-legacy: 404',
-          'immutable-chunk: 200',
-          'immutable-chunk-legacy: 200',
-          'polyfills-legacy: 404',
+          'index: text/html; charset=utf-8',
+          'index-legacy: text/html; charset=utf-8',
+          'chunk-async: text/html; charset=utf-8',
+          'chunk-async-legacy: text/html; charset=utf-8',
+          'immutable-chunk: application/javascript',
+          'immutable-chunk-legacy: application/javascript',
+          'polyfills-legacy: text/html; charset=utf-8',
         ].join('\n')
       : [
-          'index: 404',
-          'index-legacy: 404',
-          'chunk-async: 404',
-          'chunk-async-legacy: 404',
-          'immutable-chunk: 404',
-          'immutable-chunk-legacy: 404',
-          'polyfills-legacy: 404',
+          'index: text/html',
+          'index-legacy: text/html',
+          'chunk-async: text/html',
+          'chunk-async-legacy: text/html',
+          'immutable-chunk: text/html',
+          'immutable-chunk-legacy: text/html',
+          'polyfills-legacy: text/html',
```

Comment:
sirv adds `; charset=utf-8` to html files (https://github.com/lukeed/sirv/pull/122). But in `vite dev`, Vite doesn't.
Maybe we should add it?
I didn't reproduce Firefox/Chromium showing a warning described in https://github.com/lukeed/sirv/issues/106 though.

Reply:
I remember seeing some issues regarding charsets too, but I think it's better to leave no charset in dev. Not all prod servers at `; charset=utf-8` and some might not want that, so it would be safer for us to not decide on behalf.